Is_Binary_Search_Tree/Is_Binary_Search_Tree.java

```diff
@@ -0,0 +1,86 @@
+    
+    import java.util.Scanner;
+
+    public class BinaryTree {
+        class Node {
+	    int data;
+	    Node left;
+	    Node right;
+	}
+    
+	private Node root;
+	private int size;
+	public static Scanner scn = new Scanner(System.in);
+
+	public BinaryTree() {
+	    this.root = this.takeInput(null, false);
+	}
+
+	public Node takeInput(Node parent, boolean ilc) {
+
+	    int cdata = scn.nextInt();
+	    Node child = new Node();
+	    child.data = cdata;
+	    this.size++;
+
+	    // left
+	    boolean hlc = scn.nextBoolean();
+
+	    if (hlc) {
+		child.left = this.takeInput(child, true);
+	    }
+
+	    // right
+	    boolean hrc = scn.nextBoolean();
+  
+            if (hrc) {
+	        child.right = this.takeInput(child, false);
+	    }
+
+	    // return
+	    return child;
+	}
+
+	public boolean isBST() {
+	    return isBST(root);
+	}
+
+	private boolean isBST(Node node) {
+
+	    if (node == null)
+		return true;
+
+	    boolean l = isBST(node.left); 
+	    // l will store whether the left subtree is BST or not
+	    boolean r = isBST(node.right); 
+	    // r will store whether the right subtree is BST or not
+
+	    //the below given conditions will return false if the current node doesnt satisfy the property of a BST
+	    if(node.left != null && node.data < node.left.data)
+	        return false;
+
+	    if(node.right != null && node.data > node.right.data)
+		return false;
+
+	    //if the curr node satifies the BST property then we will return true if both l and r subtree are BST or not
+	    return l && r;
+	}
+
+	public static void main(String[] args) {
+
+	    BinaryTree bt = new BinaryTree();
+	    System.out.println(bt.isBST());
+	}
+    }
+
+   /*Sample Input :
+     4 true 2 true 1 false false true 3 false false true 6 true 5 false false true 7 false false
```

Comment:
Can you please re-write the code to take more understandable input format. The input format like this is making it tough to visualize the tree structure you are giving as your input.

Reply:
Should I accept the tree in level order?

Reply:
Any standard format is good.

Reply:
I saw the code for this question in c++ that has been accepted as a solution in this repository. In that file the tree has been hard cored, I too am doing the same for now.